DSharpPlus.Commands/Processors/TextCommands/TextCommandProcessor.cs

```diff
@@ -360,4 +318,95 @@ await this.extension.commandErrored.InvokeAsync(converterContext.Extension, new
+    public bool TryGetCommand(string commandText, ulong guildId, out int index, [NotNullWhen(true)] out Command? command)
+    {
+        // Declare the index here for scope, keep reading until a whitespace character is found.
+        for (index = 0; index < commandText.Length; index++)
+        {
+            if (char.IsWhiteSpace(commandText[index]))
+            {
+                break;
+            }
+        }
+
+        string rootCommandText = commandText[..index];
+        if (!this.commands.TryGetValue(rootCommandText, out command))
+        {
+            // Search for any aliases
+            foreach (Command officialCommand in this.commands.Values)
+            {
+                TextAliasAttribute? aliasAttribute = officialCommand.Attributes.OfType<TextAliasAttribute>().FirstOrDefault();
+                if (aliasAttribute is not null && aliasAttribute.Aliases.Any(alias => this.Configuration.CommandNameComparer.Equals(alias, rootCommandText)))
+                {
+                    command = officialCommand;
+                    return true;
+                }
+            }
+        }
+
+        // No alias was found
+        if (command is null || (command.GuildIds.Count > 0 && !command.GuildIds.Contains(guildId)))
+        {
+            return false;
+        }
+
+        // If there is a space after the command's name, skip it.
+        if (index < commandText.Length && commandText[index] == ' ')
+        {
+            index++;
+        }
+
+        // Recursively resolve subcommands
+        int nextIndex = index;
+        while (nextIndex != -1)
+        {
+            // If the index is at the end of the string, break
+            if (nextIndex >= commandText.Length)
+            {
+                break;
+            }
+
+            // If there was no space found after the subcommand, break
+            nextIndex = commandText.IndexOf(' ', nextIndex);
+            if (nextIndex == -1)
+            {
+                // No more spaces. Search the rest of the string to see if there is a subcommand that matches.
+                nextIndex = commandText.Length;
+            }
+
+            // Resolve subcommands
+            string subcommandName = commandText[index..nextIndex];
+
+            // Try searching for the subcommand by name, then by alias
+            // We prioritize the name over the aliases to avoid a poor dev debugging experience
+            Command? foundCommand = command.Subcommands.FirstOrDefault(subCommand => subCommand.Name.Equals(subcommandName, StringComparison.OrdinalIgnoreCase));
+            if (foundCommand is null)
+            {
+                // Search for any aliases that the subcommand may have
+                foundCommand = command.Subcommands.FirstOrDefault(subCommand => subCommand.Attributes.OfType<TextAliasAttribute>().FirstOrDefault()?.Aliases.Any(alias => this.Configuration.CommandNameComparer.Equals(alias, subcommandName)) ?? false);

```

Comment:
over-length line; what in the world is this LINQ

Reply:
Ditto, I would recommend breaking this across multiple lines (and moreover, avoiding LINQ when possible anyway) to improve readability - even with the comment, I struggle to follow all the logic